src/retrace/retrace.py

```diff
@@ -477,66 +472,92 @@ def is_package_known(package_nvr, arch, releaseid=None):
 
     return any([os.path.isfile(f) for f in candidates])
 
-# tricky
-# crash is not able to process the vmcore from different arch
-# (not even x86_64 and x86). In addition, there are several
-# types of vmcores: running 'file' command on el5- vmcore results
-# into an expected description (x86-64 or 80386 coredump), while
-# el6+ vmcores are just proclaimed 'data'. Another thing is that
-# the OSRELEASE in the vmcore sometimes contains architecture
-# and sometimes it does not.
+
+#
+# In real-world testing, approximately 60% of the time the kernel
+# version of a vmcore can be identified with the crash tool.
+# In the other 40% of the time, we must use some other method.
+#
+# The below function contains a couple regex searches that work
+# across a wide variety of vmcore formats and kernel versions.
+# We do not attempt to identify the file type since this is often not
+# reliable, but we assume the version information exists in some form
+# in the first portion of the file.  Testing has indicated that we do
+# not need to scan the entire file but can rely on a small portion
+# at the start of the file, which helps preserve useful pages in the
+# OS page cache.
+#
+# The following regex's are used for the 40% scenario
+# 1. Look for 'OSRELEASE='.  For example:
+# OSRELEASE=2.6.18-406.el5
+# NOTE: We can get "OSRELEASE=%" so we disallow the '%' character after the '='
+OSRELEASE_VAR_PARSER = re.compile("OSRELEASE=([^%][^\x00\s\n\r]+)")
+# 2. Look for "Linux version" string.  For exmaple:
+# Linux version 3.10.0-693.11.1.el7.x86_64 (mockbuild@x86-041.build.eng.bos.redhat.com) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-16) (GCC) ) #1 SMP Fri Oct 27 05:39:05 EDT 2017
+LINUX_VERSION_PARSER = re.compile('Linux\sversion\s(\S+)\s+(.*20\d{1,2}|#1\s.*20\d{1,2})')
+# 3. Look for the actual kernel release. For example:
+# 2.6.32-209.el6.x86_64 | 2.6.18-197.el5
+KERNEL_RELEASE_PARSER = re.compile('([0-9]+\.[0-9]+\.[0-9]+)-([0-9]+\.[^\x00\s\n\r]+)')
```

Comment:
Wouldn't it be better readable to use \d instead of [0-9]? Also same about [^\x00\s\n\r] as above.

Reply:
I don't think it matters but I can easily change this.

Reply:
`KERNEL_RELEASE_PARSER = re.compile('(\d+\.\d+\.\d+)-(\d+\.[^\x00\s]+)')`

Reply:
ok sure